test/cppsuite/test_harness/workload/workload_validation.h

```diff
@@ -75,66 +81,80 @@ class workload_validation {
          * checked in check_reference.
          */
         for (auto const &it : deleted_collections) {
-            if (!verify_collection_state(session, it, false))
+            if (!verify_collection_file_state(session, it, false))
                 testutil_die(DEBUG_ERROR,
-                  "validate: collection %s present on disk while it has been tracked as deleted.",
+                  "Validation failed: collection %s present on disk while it has been tracked as "
+                  "deleted.",
                   database::build_collection_name(it).c_str());
         }
 
+        /*
+         * All collections in memory should match those created in the schema tracking table.
+         * Dropping is currently not supported.
+         */
+        std::sort(created_collections.begin(), created_collections.end());
+        auto on_disk_collection = created_collections.begin();
+        for (const auto &it : known_collection_ids) {
+            if (it != *on_disk_collection)
```

Comment:
Why is it `*on_disk_collection` and not `on_disk_collection` ?

Reply:
Because it's an iterator. On a related note, you'll have to check for the end iterator before you dereference like this, otherwise, it's UB if you reach the end.

Reply:
My bad, thanks @tetsuo-cpp

Reply:
Added a size check to see that they match to address this comment and the other one by Alex C, https://github.com/wiredtiger/wiredtiger/pull/6713#discussion_r660201215.

8153016